src/Symfony/Component/Console/DependencyInjection/AddConsoleCommandPass.php

```diff
@@ -55,7 +56,7 @@ public function process(ContainerBuilder $container)
 
             $commandId = 'console.command.'.strtolower(str_replace('\\', '_', $class));
 
-            if (!isset($tags[0]['command'])) {
+            if (!isset($tags[0]['command']) && !$r->implementsInterface(DefaultNameProviderInterface::class)) {
```

Comment:
Sure. Just saying we can then cleanly wire commands, the command loader etc. The benefit would be its more flexible and allows injecting the application elsewhere... that may be useful.